### PR TITLE
Calendar availability

### PR DESCRIPTION
**Pull request by p-guelfi**
I used flatpickr.js to implement the calendar. 
In order to make it work when selecting a range of dates, I had to change the schema and replace start_date and end_date attributes of Booking with dates_range. 
You can try it out by running rails db:seed, then go to /instruments and try to book one. It will create a Booking instance, only with the dates_range for now.

### CODE REVIEW

**Comment by moelgazzar** [comment]
LGTM! 🙌

**Comment by moelgazzar** [comment]
Will adjust the DB Schema to have `dates_range`

File: app/models/booking.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class Booking < ApplicationRecord
   belongs_to :instrument
   belongs_to :user
+
+  enum status: { pending: 1, confirmed: 2, canceled: 3 }
```
Comment (DavidDicke) [inline app/models/booking.rb]: Great, you already did that. Should we have a completed status to trigger the review user flow?

File: app/views/instruments/show.html.erb [hunk 2]
```diff
@@ -3,6 +3,12 @@
 
 <h1><%= @instrument.name %></h1>
 <p><%= @instrument.description %></p>
-<p>Capacity: <%= @instrument.capacity %></p>
 <p>Price: <%= @instrument.price %></p>
 <%= link_to 'Back to Instruments', instruments_path %>
+
+
+<h2>Book</h2>
```
Comment (DavidDicke) [inline app/views/instruments/show.html.erb:10]: I think we need to have a gate here to only allow logged in users to book. I thought easiest would thus be a booking new page that can only be reached by users. So there could be a link here and the form on the bookings new page.

**Review by DavidDicke — COMMENTED** [review]
Hey Pablo, I added some comments, but I did not watch the lecture yet. (I will do it now) So I can't give feedback to flatpickr.js Otherwise looks great!

File: app/models/booking.rb (hunk 1 above)
Comment (p-guelfi) [reply app/models/booking.rb]: Yes, good idea

File: app/views/instruments/show.html.erb (hunk 2 above)
Comment (p-guelfi) [reply app/views/instruments/show.html.erb:10]: Yes, that's how I have it on this branch, after seeing your user flow on Figma. With "before_action :authenticate_user!" on the application controller and a "skip_before_action :authenticate_user!, only: [:index, :show]" on the instruments controller. If you click on "book" you are prompted to login before you see the booking new page.

**Comment by DavidDicke** [comment]
Hey Pablo, great work again. If it works I have nothing against merging.

**Pull request merged**